reproject/mosaicking/subset_array.py

```diff
@@ -73,15 +76,19 @@ def _operation(self, other, op):
 
         # Extract cutout from each
 
-        self_array = self.array[jmin - self.jmin:jmax - self.jmin,
-                                imin - self.imin:imax - self.imin]
-        self_footprint = self.footprint[jmin - self.jmin:jmax - self.jmin,
-                                        imin - self.imin:imax - self.imin]
-
-        other_array = other.array[jmin - other.jmin:jmax - other.jmin,
-                                  imin - other.imin:imax - other.imin]
-        other_footprint = other.footprint[jmin - other.jmin:jmax - other.jmin,
-                                          imin - other.imin:imax - other.imin]
+        self_array = self.array[
+            jmin - self.jmin : jmax - self.jmin, imin - self.imin : imax - self.imin
+        ]
+        self_footprint = self.footprint[
+            jmin - self.jmin : jmax - self.jmin, imin - self.imin : imax - self.imin
+        ]
+
+        other_array = other.array[
+            jmin - other.jmin : jmax - other.jmin, imin - other.imin : imax - other.imin
+        ]
+        other_footprint = other.footprint[
+            jmin - other.jmin : jmax - other.jmin, imin - other.imin : imax - other.imin
+        ]
```

Comment:
I am unconvinced this is an improvement. Might we worth ignoring?

Reply:
Yup, this is a nice example where ignoring would be a good thing. But I might suggest to go half-way to the black style with:
```
self_array = self.array[
    jmin - self.jmin:jmax - self.jmin,
    imin - self.imin:imax - self.imin
]
```

Reply:
I've recently acquired a real distaste for the hanging aligned list like in the original. It just looks messy, especially since the alignment is different in each of the four cases.

Reply:
Adding an extra comma at the end would force black to format on multiple lines (as suggested by @taldcroft):
```
self_array = self.array[
    jmin - self.jmin:jmax - self.jmin,
    imin - self.imin:imax - self.imin,
]
```
which takes a bit more space but I think is a good compromise.

Reply:
> Adding an extra comma at the end

that's really stupid.

Reply:
Well it's useful in some cases, such as the current one.
https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma

Reply:
That's too magic for my taste.

Reply:
It makes you feel like the sentence isn't quite finished,

Reply:
Just for the record, if we are talking about who likes what, I find the hanging indented list rather helpful and love those compared to the distasteful output of black (noone wrote code like that by themselves, for good reasons), it very much drives the eye to the variable they belong to. 
Unlike a single closing paren on a new line, or trailing comas which only purpose is to hack a tool that is brute forcing its way even when it makes no sense.